neo4j/v1/session.py

```diff
@@ -341,6 +355,11 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_value, traceback):
         self.close()
 
+    def reset(self):
```

Comment:
Should reset really be a user-facing feature? I'd argue we should encourage users of the Session API to simply close sessions if they are unclear about their state, unless there's a good use case that warrants doing this instead?